Airwallex/Core/Sources/Internal/AWXPaymentMethodListViewController.m

```diff
@@ -141,11 +141,15 @@ - (void)loadAvailablePaymentConsents {
     if ([self.session isKindOfClass:[AWXOneOffSession class]] && customerPaymentConsents.count > 0 && customerPaymentMethods.count > 0) {
         NSArray *paymentConsents = [customerPaymentConsents filteredArrayUsingPredicate:[NSPredicate predicateWithFormat:@"nextTriggeredBy == %@ AND status == 'VERIFIED'", FormatNextTriggerByType(AirwallexNextTriggerByCustomerType)]];
         NSMutableArray *availablePaymentConsents = [@[] mutableCopy];
+        NSMutableArray *cardsFingerPrint = [NSMutableArray new];
         for (AWXPaymentConsent *consent in paymentConsents) {
             AWXPaymentMethod *paymentMethod = [customerPaymentMethods filteredArrayUsingPredicate:[NSPredicate predicateWithFormat:@"Id == %@", consent.paymentMethod.Id]].firstObject;
             if (paymentMethod != nil) {
-                consent.paymentMethod = paymentMethod;
-                [availablePaymentConsents addObject:consent];
+                if (![cardsFingerPrint containsObject:paymentMethod.card.fingerprint]) {
```

Comment:
Oh man, this should really be a backend thing. Does web do this on frontend as well?

Reply:
yes frontend as well, it's PA's decision since they said backend can't have a quick fix